iree/base/loop.c

```diff
@@ -0,0 +1,203 @@
+// Copyright 2022 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/base/loop.h"
+
+#include "iree/base/tracing.h"
+
+//===----------------------------------------------------------------------===//
+// iree_loop_t
+//===----------------------------------------------------------------------===//
+
+IREE_API_EXPORT iree_status_t iree_loop_call(iree_loop_t loop,
+                                             iree_loop_priority_t priority,
+                                             iree_loop_callback_fn_t callback,
+                                             void* user_data) {
+  if (IREE_UNLIKELY(!loop.ctl)) {
+    return iree_make_status(IREE_STATUS_INVALID_ARGUMENT, "null loop");
+  }
```

Comment:
I'll do a scrub and see how much of that I need

Reply:
I think they're all needed here - we want to error out when the user tries to do things with the null loop and make sure that the user knows that the work they requested is never going to execute - otherwise it's easy to fire off requests that clean up resources that never actually clean things up.